index.ts

```diff
@@ -7,19 +7,45 @@ string,
 Promise<browser.contentScripts.RegisteredContentScript>
 >();
 
-type ContentScripts = NonNullable<chrome.runtime.Manifest['content_scripts']>;
+const chromeRegister = globalThis?.chrome?.scripting?.registerContentScripts;
+const firefoxRegister = globalThis?.browser?.contentScripts?.register;
 
-const registerContentScript
-	= globalThis?.browser?.contentScripts?.register
-	?? registerContentScriptPonyfill;
+async function registerContentScript(
+	contentScript: ChromeContentScript,
+): Promise<browser.contentScripts.RegisteredContentScript> {
+	if (chromeRegister) {
+		const id = 'webext-dynamic-content-script-' + JSON.stringify(contentScript);
+		await chromeRegister([{
+			id,
+			...contentScript,
+		}]);
```

Comment:
Since there's no such thing as `onStartup` that consistently runs when the extension is enabled, I think the easiest solution is to just attempt to register the script and catch the error. Script registration is already de-duplicated by key, so this is likely the safest way.

- [ ] Add `.catch(ignoreIfDuplicateError)`